### PR TITLE
 Allow to publish local address to local zookeeper only

### PR DESCRIPTION
**Pull request by jitendradhawan**

### CODE REVIEW

File: src/main/java/io/appform/dropwizard/discovery/bundle/ServiceDiscoveryBundle.java [hunk 1]
```diff
@@ -188,11 +195,33 @@ protected int getPort(T configuration) {
...
-                    .getCanonicalHostName();
-        }
+        val host = (Strings.isNullOrEmpty(serviceDiscoveryConfiguration.getPublishedHost())
+                || serviceDiscoveryConfiguration.getPublishedHost()
+                .equals(Constants.DEFAULT_HOST))
+                   ? InetAddress.getLocalHost()
+                           .getCanonicalHostName()
+                   : serviceDiscoveryConfiguration.getPublishedHost();
+
+        String publishedHostAddress = InetAddress.getByName(host)
```
Comment (santanusinha) [inline src/main/java/io/appform/dropwizard/discovery/bundle/ServiceDiscoveryBundle.java:205]: Be consistent with val/var

File: src/main/java/io/appform/dropwizard/discovery/bundle/ServiceDiscoveryConfiguration.java [hunk 2]
```diff
@@ -78,21 +86,27 @@ public class ServiceDiscoveryConfiguration {
 
     private Set<String> tags;
 
+    @JsonIgnore
+    public Set<String> getZookeeperHosts() {
+        return Arrays.stream(zookeeper.split(ZOOKEEPER_HOST_DELIMITER))
+                .map(zkHostPort -> zkHostPort.split(HOST_PORT_DELIMITER)[0])
```
Comment (santanusinha) [inline src/main/java/io/appform/dropwizard/discovery/bundle/ServiceDiscoveryConfiguration.java:92]: Can you check if a path can be provided in this conn string? if yes, those need to be removed before parsing

File: src/test/java/io/appform/dropwizard/discovery/bundle/ServiceDiscoveryBundleLocalHostPortTest.java [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+    private final HealthCheckRegistry healthChecks = mock(HealthCheckRegistry.class);
+    private final JerseyEnvironment jerseyEnvironment = mock(JerseyEnvironment.class);
+    private final MetricRegistry metricRegistry = mock(MetricRegistry.class);
+    private final LifecycleEnvironment lifecycleEnvironment = new LifecycleEnvironment(metricRegistry);
+    private final Environment environment = mock(Environment.class);
+    private final Bootstrap<?> bootstrap = mock(Bootstrap.class);
+    private final Configuration configuration = mock(Configuration.class);
+
+    static {
+        val root = (Logger) LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
```
Comment (santanusinha) [inline src/test/java/io/appform/dropwizard/discovery/bundle/ServiceDiscoveryBundleLocalHostPortTest.java:63]: setup logback-testing.xml and remove this

File: src/test/java/io/appform/dropwizard/discovery/bundle/ServiceDiscoveryBundleLocalHostPortTest.java [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+
+        });
+
+        assertTrue(thrown.getMessage()
+                .contains("Not allowed to publish localhost address to remote zookeeper"));
+
+    }
+
+    @Test
+    void shouldThrowExceptionWhenUsingUnresolvableZkHost() {
```
Comment (santanusinha) [inline src/test/java/io/appform/dropwizard/discovery/bundle/ServiceDiscoveryBundleLocalHostPortTest.java:127]: Rename these methods to smaller names > Junit 5 supports `@DisplayName`, please use that

**Review by santanusinha — CHANGES_REQUESTED** [review]

File: src/main/java/io/appform/dropwizard/discovery/bundle/ServiceDiscoveryBundle.java (hunk 1 above)
Comment (jitendradhawan) [reply src/main/java/io/appform/dropwizard/discovery/bundle/ServiceDiscoveryBundle.java:205]: replaced with val

File: src/test/java/io/appform/dropwizard/discovery/bundle/ServiceDiscoveryBundleLocalHostPortTest.java (hunk 4 above)
Comment (jitendradhawan) [reply src/test/java/io/appform/dropwizard/discovery/bundle/ServiceDiscoveryBundleLocalHostPortTest.java:127]: renamed methods

File: src/main/java/io/appform/dropwizard/discovery/bundle/ServiceDiscoveryConfiguration.java (hunk 2 above)
Comment (jitendradhawan) [reply src/main/java/io/appform/dropwizard/discovery/bundle/ServiceDiscoveryConfiguration.java:92]: Splitting on "/" and taking the hostname now

File: src/test/java/io/appform/dropwizard/discovery/bundle/ServiceDiscoveryBundleLocalHostPortTest.java (hunk 3 above)
Comment (jitendradhawan) [reply src/test/java/io/appform/dropwizard/discovery/bundle/ServiceDiscoveryBundleLocalHostPortTest.java:63]: removed

**Review by santanusinha — APPROVED** [review]

**Review by santanusinha — APPROVED** [review]

**Pull request merged**